### PR TITLE
Ensure that only correct users can access year/module/lesson pages

### PR DESCRIPTION
**Pull request by callumacrae**
### Context

from the backlog of michal 🧠 (aka no ticket)

### Testing

yes 😩

### CODE REVIEW

File: app/controllers/core_induction_programmes/lesson_parts_controller.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ class CoreInductionProgrammes::LessonPartsController < ApplicationController
   include CipBreadcrumbHelper
 
   after_action :verify_authorized
-  before_action :authenticate_user!, except: :show
+  before_action :authenticate_user!
```
Comment (kicferk1) [inline app/controllers/core_induction_programmes/lesson_parts_controller.rb:9]: Good spot!

File: spec/policies/course_lesson_policy_spec.rb [hunk 2]
```diff
@@ -4,30 +4,51 @@
...
+      ect_user = create(:user, :early_career_teacher, { core_induction_programme: cip_for_lesson })
+      user = create(:user, :mentor)
+      user.mentor_profile.early_career_teachers = [ect_user]
+      user
+    end
+    it { is_expected.to permit_action(:show) }
+    it { is_expected.to forbid_edit_and_update_actions }
+    it { is_expected.to forbid_action(:update_progress) }
+  end
+  context "mentor without access" do
```
Comment (kicferk1) [inline spec/policies/course_lesson_policy_spec.rb:43]: Let's add spaces after `end`s - I think they make things easier to read

**Review by kicferk1 — APPROVED** [review]
Looks good, let's add a little bit of white-space in tests, empty lines after `end`s really help figuring out what you are looking at, at least for me :)

**Comment by kicferk1** [comment]
Looks like a cypress meta tag relied on being able to access module page without authentication 😂

**Review by kicferk1 — APPROVED** [review]
